### PR TITLE
Cleanup and additional tests for ConfigBuilder

### PR DESCRIPTION
**Pull request by thisthat**
This PR removes useless code and adds more tests for `ConfigBuilder` classes

### CODE REVIEW

File: sdk/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker [hunk 1]
```diff
@@ -0,0 +1 @@
+mock-maker-inline
```
Comment (bogdandrutu) [inline sdk/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:1]: Can you comment somewhere why is this needed? I am sure we will forget.

File: sdk/src/test/java/io/opentelemetry/sdk/common/export/ConfigBuilderTest.java [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+  @Test
+  public void testNormalize_dotMap() {
+    Map<String, String> map = new HashMap<>();
+    map.put("lower.case", "1");
+    map.put("lower_case", "2");
+    map.put("loWer.cAsE", "3");
+    map.put("loWer_cAsE", "4");
+    Map<String, String> normalized = NamingConvention.DOT.normalize(map);
+    assertThat(normalized.size()).isEqualTo(2);
+    assertThat(normalized.get("lower.case")).isNotEmpty();
```
Comment (bogdandrutu) [inline sdk/src/test/java/io/opentelemetry/sdk/common/export/ConfigBuilderTest.java:62]: Use containsExactly to confirm that only expected values are in the map

File: sdk/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker (hunk 1 above)
Comment (thisthat) [reply sdk/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:1]: I created a comment in `[Simple|Batch]SpanProcessorTest`

**Comment by carlosalberto** [comment]
@thisthat Please rebase ;)

**Comment by thisthat** [comment]
@carlosalberto done! :)

File: sdk/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker (hunk 1 above)
Comment (carlosalberto) [reply sdk/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:1]: @thisthat I'm guessing we can't create a comment right in this file?

Comment (thisthat) [reply sdk/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:1]: Unfortunately, according to the [documentation](https://github.com/mockito/mockito/wiki/What%27s-new-in-Mockito-2#mock-the-unmockable-opt-in-mocking-of-final-classesmethods) it says "containing a single line" :(

File: sdk/src/test/java/io/opentelemetry/sdk/common/export/ConfigBuilderTest.java [hunk 3]
```diff
@@ -123,4 +127,174 @@ public void nullValue_DoubleProperty() {
...
+
+  @Test
+  public void testNormalize_dotMap() {
+    Map<String, String> map = new HashMap<>();
+    map.put("lower.case", "1");
+    map.put("lower_case", "2");
+    map.put("loWer.cAsE", "3");
+    map.put("loWer_cAsE", "4");
+    Map<String, String> normalized = NamingConvention.DOT.normalize(map);
+    assertThat(normalized.size()).isEqualTo(2);
```
Comment (bogdandrutu) [inline sdk/src/test/java/io/opentelemetry/sdk/common/export/ConfigBuilderTest.java:155]: This is not necessary if you use contains exactly :)
